cmd/erasure-object.go

```diff
@@ -1263,7 +1264,8 @@ func (er erasureObjects) putObject(ctx context.Context, bucket string, object st
 	}
 
 	userDefined["etag"] = r.MD5CurrentHexString()
-	if opts.PreserveETag != "" {
+	_, ok := crypto.IsEncrypted(userDefined)
```

Comment:
I think S3.IsEncrypted() is better.

We can keep ETags for non encrypted, SSE-KMS, multipart but not SSE-S3

Reply:
multipart will never come here @vadmeste for preserveETag and in-case of SSE-KMS also we must save the longer encrypted version.

That's why this code says that it must be plain-text only to be able to preserve a custom etag remotely.